kpi-tracker/src/main.rs

```diff
@@ -0,0 +1,229 @@
+use std::collections::{BTreeMap, HashMap};
+use std::hash::Hash;
+
+use anyhow::Context;
+use chrono::{DateTime, Utc};
+use clap::Parser;
+use concordium_rust_sdk::{
+    id::types::AccountCredentialWithoutProofs,
+    smart_contracts::common::AccountAddress,
+    types::{
+        hashes::{BlockMarker, HashBytes},
+        AbsoluteBlockHeight,
+    },
+    v2::{self, AccountIdentifier, Client, Endpoint, FinalizedBlockInfo},
+};
+use futures::{self, StreamExt};
+
+#[derive(Debug, Parser)]
+struct Args {
+    /// The node used for querying
+    #[arg(long, default_value = "http://localhost:20001")]
+    node: Endpoint,
+    /// How many blocks to process.
+    // Only here for testing purposes...
+    #[arg(long, default_value_t = 10000)]
+    num_blocks: u64,
+}
+
+#[derive(Debug)]
+struct AccountDetails {
+    is_initial: bool,
+    block_hash: String, // FK to blocks
```

Comment:
Please use proper types. So `BlockHash` here. No strings!

Reply:
I wanted to do this, but I also need to use this for a key in `HashMap<BlockHash, BlockDetails>`, where I get an error that the trait bounds of the hashmap are not satisfied, as the key needs to implement `std::hash::Hash` which `BlockHash` doesn't seem to do.

What would you do here?

Reply:
I would add a Hash implementation in base.